modules/javafx.graphics/src/main/java/com/sun/glass/ui/gtk/screencast/ScreencastHelper.java

```diff
@@ -0,0 +1,233 @@
+    }
+
+    private static synchronized native void closeSession();
+
+    private static void timerCloseSessionRestart() {
+        if (timerTask != null) {
+            timerTask.cancel();
+        }
+
+        timerTask = new TimerTask() {
+            @Override
+            public void run() {
+                closeSession();
+            }
+        };
+
+        timerCloseSession.schedule(timerTask, DELAY_BEFORE_SESSION_CLOSE);
+    }
+
+    public static synchronized void getRGBPixels(
+            int x, int y, int width, int height, int[] pixelArray
+    ) {
+        if (!IS_NATIVE_LOADED) return;
+
+        timerCloseSessionRestart();
+
+        Rectangle captureArea = new Rectangle(x, y, width, height);
+
+        List<Rectangle> affectedScreenBounds = getSystemScreensBounds()
+                .stream()
+                .filter(captureArea::intersects)
+                .toList();
+
+        if (SCREENCAST_DEBUG) {
+            System.out.printf("// getRGBPixels in %s, affectedScreenBounds %s\n",
+                    captureArea, affectedScreenBounds);
+        }
+
+        if (affectedScreenBounds.isEmpty()) {
+            if (SCREENCAST_DEBUG) {
+                System.out.println("// getRGBPixels - requested area "
+                        + "outside of any screen");
+            }
+            return;
+        }
+
+        int retVal;
+        Set<TokenItem> tokensForRectangle =
+                TokenStorage.getTokens(affectedScreenBounds);
+
+        int[] affectedScreenBoundsArray = affectedScreenBounds
+                .stream()
+                .filter(captureArea::intersects)
+                .flatMapToInt(bounds -> IntStream.of(
+                        bounds.x, bounds.y,
+                        bounds.width, bounds.height
+                ))
+                .toArray();
+
+        for (TokenItem tokenItem : tokensForRectangle) {
+            retVal = getRGBPixelsImpl(
+                    x, y, width, height,
+                    pixelArray,
+                    affectedScreenBoundsArray,
+                    tokenItem.token
+            );
+
+            if (retVal >= 0) { // we have received a screen data
+                return;
+            } else if (!checkReturnValue(retVal)) {
+                return;
+            } // else, try other tokens
+        }
+
+        // we do not have a saved token or it did not work,
+        // try without the token to show the system's permission request window
+        retVal = getRGBPixelsImpl(
+                x, y, width, height,
+                pixelArray,
+                affectedScreenBoundsArray,
+                null
+        );
+
+        checkReturnValue(retVal);
+    }
+
+    private static boolean checkReturnValue(int retVal) {
+        if (retVal == DENIED) {
+            // user explicitly denied the capture, no more tries.
+            throw new SecurityException(
+                    "Screen Capture in the selected area was not allowed"
+            );
```

Comment:
@kevinrushforth 
I think this SecurityException should be removed, as it is not mentioned in the javadoc for the `getPixelColor` or `getScreenCapture`.  And we also do not want to throw another runtime exception.

Reply:
If the SecurityException is not thrown, how does the caller of `getRGBPixels` knows that the result will not be correct because of the missing permission? (as opposed to being incorrect due to a failure)

Reply:
> If the SecurityException is not thrown, how does the caller of getRGBPixels knows that the result will not be correct because of the missing permission? (as opposed to being incorrect due to a failure)

He won't know it. An indirect result of failure can be a black image.
The [current documentation](https://github.com/openjdk/jfx/blob/master/modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java#L240) documentation specifies only IllegalStateException and NullPointerException.

Reply:
Since `SecurityException` is deprecated, we should not throw that.

In a similar situation on macOS (user has not enabled screen capture), we return a blank (all black) image, so I think that's what we would do here.

Reply:
Updated to return blank image in case of permission denial.

Reply:
> Since `SecurityException` is deprecated, we should not throw that.
> 
> In a similar situation on macOS (user has not enabled screen capture), we return a blank (all black) image, so I think that's what we would do here.

No, SecurityException is NOT deprecated and there are no plans to deprecate it.

https://download.java.net/java/early_access/jdk23/docs/api/java.base/java/lang/SecurityException.html#%3Cinit%3E()

This kind of platform reason for a "SecurityException" is precisely why it isn't deprecated even though the Java SecurityManager is deprecated.

Reply:
Thanks for the correction about "SecurityException" not being deprecated.

I still think that for this PR, we should not throw an exception, since we A) don't specify it to do so, and B) don't do that on macOS in a nearly identical situation.

We could consider a future RFE to change the spec (and maybe the behavior).